src/commands/cmd_list.cc

```diff
@@ -154,6 +154,79 @@ class CommandRPop : public CommandPop {
   CommandRPop() : CommandPop(false) {}
 };
 
+class CommandLMPop : public Commander {
+ public:
+  /// format: LMPOP #numkeys key0 [key1 ...] <LEFT | RIGHT> [COUNT count]
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 1);
+
+    auto num_keys = GET_OR_RET(parser.TakeInt<uint32_t>());
+    keys_.clear();
+    keys_.reserve(num_keys);
+    for (uint32_t i = 0; i < num_keys; ++i) {
+      keys_.emplace_back(GET_OR_RET(parser.TakeStr()));
+    }
+
+    auto left_or_right = util::ToLower(GET_OR_RET(parser.TakeStr()));
+    if (left_or_right == "left") {
+      left_ = true;
+    } else if (left_or_right == "right") {
+      left_ = false;
+    } else {
+      return {Status::RedisParseErr, errInvalidSyntax};
+    }
+
+    if (parser.Good()) {
```

Comment:
i prefer using the while loop (though we now only have one arg), so that we can remove the parser.Good() in `return parser.Good() ? parser.InvalidSyntax() : Status::OK();`
```suggestion
    while (parser.Good()) {
```

Reply:
I guess we don't need `while` because if that we need to check the count of args later?

Reply:
Please refer to the discussion of bulk, pragmaTwice and me above. 

The while loop has no order semantic, and cannot guarantee the orders of keywords. However, as aligned to redis behavior, the LMPOP's keywords MUST have strict order. So I choose to not to adopt the CommandParser's while() loop schema.

Reply:
i did see the disscussion below

> LMPOP's keywords MUST have strict order

what do you mean that Redis's LMPOP must have the strict order? The COUNT is an optional arg, it dose not require the order. what happen if Redis's LMPOP have more optional args in the future?

Reply:
the keys and the left_or_right dose require the order (and you already handle it), but the rest of the args are the optional, although now it only have one COUNT arg

Reply:
I could see your point, but the while loop here would bring another problem: what if the COUNT keyword is repeated?
For redis, the command "LMPOP 1 mylist RIGHT COUNT 1 COUNT 2" is considered invalid. If we use while loop to parse the COUNT keyword here, the repeated COUNT keywords would be accepted.
Sure we could use another flag like "is_count_parsed" to check if we have seen the COUNT before. But for what? These additional checking already brought more complexity than the "return isGood() ? caseA : caseB"!
I do try to modify the code as you suggest, but it really looks even uglier to me. If more optional keywords are added, then the parsing could be modified, but the repeat problem still exists.
I've tested other commands such as "LPOS", and their optional keywords could be repeated with only the last one works. But for LMPOP, the COUNT keyword could not be repeated. I don't know if this is a redis bug or redis feature.

Reply:
Redis's optional args for old commands are indeed, some can be repeated. But actually that is not very important, or the Redis community is not willing to spend energy to modify it, it will cost time and introduce the changes or break the git blame log

For the new commands, like LMPOP, it was introduce in Redis 7.0, or in fact, i am the one introduced it. normally, a new `is_count_parsed` is indeed ugly. In redis, the count will be init   to -1, and use something like `if (count == -1 && parser.EatEqICase("count"))` format to handle the repeat issue, and in the last, do a `if (count == -1) count = 1;` before the execute

Reply:
Refactored as suggested.